packages/perseus/src/widgets/label-image/label-image.tsx

```diff
@@ -466,11 +434,43 @@ export class LabelImage
             selected: selected.length ? selected : undefined,
         });
     }
-    renderMarkers(): ReadonlyArray<React.ReactNode> {
-        const {markers, questionCompleted, preferredPopoverDirection} =
-            this.props;
 
-        const {activeMarkerIndex, markersInteracted} = this.state;
+    /**
+     * Get marker with the appropriate selection state for current display mode.
+     * When showing solutions, it auto-selects correct answers, or clears
+     * selections for markers without answers.
+     *
+     * @param marker - The marker to update.
+     * @returns The updated marker state.
+     */
+    getUpdatedMarkerState(
+        marker: OptionalAnswersMarkerType,
+    ): OptionalAnswersMarkerType {
+        const shouldShowFeedback =
+            this.props.showSolutions === "all" || this.props.reviewMode;
```

Comment:
Given such minimal use of `this` in this function, what do you think of making it static, or even moving it out of this component completely?  I think a pure function might make this function even clearer, plus easy to test.